This really disables altivec if not wanted.
Patch from xtophe: https://trac.videolan.org/vlc/ticket/1055
https://bugs.gentoo.org/show_bug.cgi?id=158126
Causes a compile failure because altivec.h is found and -maltivec not appended to cflags
updated to try to fix this
===================================================================
Index: vlc-0.8.6_p18636/configure.ac
===================================================================
--- vlc-0.8.6_p18636.orig/configure.ac
+++ vlc-0.8.6_p18636/configure.ac
@@ -1301,10 +1301,12 @@ AC_CACHE_CHECK([if \$CC groks AltiVec C 
 if test "${ac_cv_c_altivec}" != "no"; then
   CPPFLAGS="${CPPFLAGS_save} ${ac_cv_c_altivec}"
 fi
+if test "${enable_altivec}" != "no"; then
 AC_CHECK_HEADERS(altivec.h)
+fi
 CPPFLAGS="${CPPFLAGS_save}"
 
-if test "${ac_cv_c_altivec}" != "no"; then
+if test "${ac_cv_c_altivec}" != "no" -a "${enable_altivec}" != "no"; then
   AC_DEFINE(CAN_COMPILE_C_ALTIVEC, 1, Define if your compiler groks C AltiVec extensions.)
   VLC_ADD_CFLAGS([vlc],[${ac_cv_c_altivec}])
   VLC_ADD_CFLAGS([idctaltivec motionaltivec],[${ac_cv_c_altivec}])
